.github/actions/python-setup/action.yml

```diff
@@ -0,0 +1,15 @@
+name: Setup
+description: Perform initial setup of package
+
+runs:
+  using: composite
+  steps:
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
+      with:
+         enable-cache: true
```

Comment:
This should not be necessary, since `setup-uv` enables caching by default on GitHub runners (See https://github.com/astral-sh/setup-uv/releases/tag/v5.0.0 )
```suggestion
```